### PR TITLE
Document internals of audio virtualization

### PR DESCRIPTION
**Pull request by marmarek**

### CODE REVIEW

**Comment by andrewdavidwong** [comment]
Would you like me to do anything with this, @marmarek, or are you still working on it?

**Comment by marmarek** [comment]
I'm still working on some details (trying to implement things for R4.1 according to this spec), but the general layout is done. Feel free to review.

**Comment by andrewdavidwong** [comment]
@marmarek: I've pushed some light revisions.

Sorry about all the other clutter... Is that normal when your branch isn't caught up, or should I do something different in this situation?

**Comment by marmarek** [comment]
> git pull https://github.com/marmarek/qubes-doc.git audio-internals

`git pull` do `fetch` and `merge`. To add changes to a branch without merging it with master, you should do just `fetch`, like this:
```
git fetch https://github.com/marmarek/qubes-doc.git audio-internals:marmarek-audio-internals
git checkout marmarek-audio-internals
```

File: system/audio.md [hunk 1]
```diff
@@ -1,58 +1,70 @@
...
+ - `0x00020001` -- VM does want to send audio output.
 
 pacat-simple-vchan
 ------------------
 
-Dom0 (or Audio VM) part of the audio virtualization. It is responsible for transferring audio samples between pulseaudio daemon in Dom0/Audio VM (having access to actual audio hardware) and a VM playing/recording.
-For each VM, matching `pacat-simple-vchan` process is running. Each of them open a separate input and output streams to local pulseaudio, which allows to control volume of each VM separately (using `pavucontrol` tool for example).
+This is the dom0 (or Audio VM) part of the audio virtualization.
+It is responsible for transferring audio samples between the PulseAudio daemon in dom0/Audio VM (which has access to the actual audio hardware) and a VM playing/recording sounds.
+In each VM, a corresponding `pacat-simple-vchan` process is running.
```
Comment (marmarek) [inline system/audio.md]: `pacat-simple-vchan` is running in dom0, separate for each VM having audio. Not in a VM.

**Comment by andrewdavidwong** [comment]
> > git pull https://github.com/marmarek/qubes-doc.git audio-internals
> 
> `git pull` do `fetch` and `merge`. To add changes to a branch without merging it with master, you should do just `fetch`, like this:
> 
> ```
> git fetch https://github.com/marmarek/qubes-doc.git audio-internals:marmarek-audio-internals
> git checkout marmarek-audio-internals
> ```

Got it, thanks.

File: system/audio.md (hunk 1 above)
Comment (andrewdavidwong) [reply system/audio.md]: Fixed, thanks.

**Comment by andrewdavidwong** [comment]
I went ahead and merged this because my previous mistake of pulling instead of fetching made this PR hard to follow and was causing spurious Travis errors. Plus, the main thing is that this page is a great addition to the docs in its own right that we can always continue to iterate on, so it's best to get it merged in now. (If you think that the specifics of the implementation might change significantly, we can add a notice at the top stating that it's still under development and subject to change.)
